packedForest/src/forestTypes/unsupervisedForests/urf/unprocessedURFNode.h

```diff
@@ -0,0 +1,183 @@
+#ifndef rfunprocessedURFNode_h
+#define rfunprocessedURFNode_h
+#include "splitURF.h"
+#include "../baseUnprocessedNodeUnsupervised.h"
+#include "../stratifiedInNodeClassIndicesUnsupervised.h"
+#include <vector>
+#include <random>
+#include <assert.h>
+
+namespace fp{
+
+
+	template <typename T> //
+		class unprocessedURFNode : public baseUnprocessedNodeUnsupervised<T>{
+			protected:
+
+				splitURFInfo<T> bestSplitInfo;
+				std::vector<int> featuresToTry;
+
+				//std::random_device rd;
+				//The next three should be static
+				//std::mt19937 rng(rd());   
+				//std::uniform_int_distribution<int> randomMtry(0,fpSingleton::getSingleton().returnMtry()-1);
+				//std::uniform_int_distribution<int> randomFeature(0,fpSingleton::getSingleton().returnNumFeatures()-1);
+				//Example: auto random_integer = uni(rng);
+
+			public:
+                                unprocessedURFNode(int numObsForRoot): baseUnprocessedNodeUnsupervised<T>::baseUnprocessedNodeUnsupervised(numObsForRoot), featuresToTry(fpSingleton::getSingleton().returnMtry()){}
+
+
+                                unprocessedURFNode(int parentID, int dep, bool isLeft): baseUnprocessedNodeUnsupervised<T>::baseUnprocessedNodeUnsupervised(parentID, dep, isLeft){
+}
+
+				~unprocessedURFNode(){}
+				
+
+				inline int returnBestFeature(){
+					return bestSplitInfo.returnFeatureNum();
+				}
+
+				inline double returnBestImpurity(){
+					return bestSplitInfo.returnImpurity();
+				}
+
+				inline T returnBestCutValue(){
+					return bestSplitInfo.returnSplitValue();
+				}
+
+
+				inline void setBestSplit(splitURFInfo<T> tempSplit){
+					if(tempSplit.returnImpurity() >= 0){
```

Comment:
same question as before, why do we need to check that tempSplit >= 0?

Reply:
This is because in splitURerF.h (inside the function splitURerFInfo<T> twoMeanSplit), we set the impurity to -1 if all the feature values are equal or if there aren't any nonzero values. In those cases, we do NOT split and instead set the node to a leaf node. Thus we are checking to make sure the impurity >= 0 and split only if that holds.